packages/functional-tests/pages/signupReact.ts

```diff
@@ -1,118 +1,134 @@
-    if (submit) {
-      await this.submit('Create account');
-    }
+  get cannotCreateAccountHeading() {
+    return this.page.getByRole('heading', { name: 'Cannot create account' });
   }
 
-  async fillOutCodeForm(email: string) {
-    const code = await this.target.email.waitForEmail(
-      email,
-      EmailType.verifyShortCode,
-      EmailHeader.shortCode
+  goto(route = '/', params = new URLSearchParams()) {
+    params.set('forceExperiment', 'generalizedReactApp');
+    params.set('forceExperimentGroup', 'react');
+    return this.page.goto(
+      getReactFeatureFlagUrl(this.target, route, params.toString())
     );
-    await this.setCode(code);
-    await this.submit('Confirm');
   }
 
-  async goToEmailFirstAndCreateAccount(
-    params: URLSearchParams,
-    email: string,
-    password: string
-  ) {
-    await this.goto('/', params);
-    // fill out email first form
-    await this.fillOutEmailFirst(email);
-    await this.page.waitForURL(/signup/);
-    await this.page.waitForSelector('#root');
-    await this.fillOutSignupForm(password);
-    await this.page.waitForURL(/confirm_signup_code/);
-    await this.fillOutCodeForm(email);
-  }
+  async fillOutEmailForm(formData: EmailFormData) {
+    await expect(
+      this.emailFormHeading,
+      'Check for email form failed. The heading missing or not as expected.'
+    ).toBeVisible();
+
+    await this.emailTextbox.fill(formData.email);
 
-  async submit(label: string) {
-    await this.page.getByRole('button', { name: label }).click();
+    if (formData.submit) {
+      await this.submitButton.click();
+    }
   }
 
-  async confirmCodeHeading() {
-    this.page.getByRole('heading', { name: /Enter confirmation code/ });
+  async fillOutSignupForm(formData: SignupFormData) {
+    await expect(
+      this.signupFormHeading,
+      'Check for signup form failed. The heading missing or not as expected.'
+    ).toBeVisible();
+
+    await this.passwordTextbox.fill(formData.password);
+    await this.verifyPasswordTextbox.fill(formData.password);
+    await this.ageTextbox.fill(formData.age);
+
+    if (formData.submit) {
+      await this.createAccountButton.click();
+    }
   }
 
-  async visitTermsOfServiceLink() {
-    const link = this.page.getByText('Terms of Service');
-    link.click();
+  async fillOutCodeForm(formData: CodeFormData) {
+    const code = await this.target.email.waitForEmail(
+      formData.email,
+      EmailType.verifyShortCode,
+      EmailHeader.shortCode
+    );
+
+    await expect(
+      this.codeFormHeading,
+      'Check for code form failed. The heading missing or not as expected.'
+    ).toBeVisible();
+
+    await this.codeTextbox.fill(code);
+
+    if (formData.submit) {
+      await this.confirmButton.click();
+    }
   }
 
-  async visitPrivacyPolicyLink() {
-    const link = this.page.getByText('Privacy Notice');
-    link.click();
+  async waitForRoot() {
+    const root = this.page.locator('#root');
+    await root.waitFor();
```

Comment:
Is this used?

Reply:
Yes, although I don't understand it's full purpose atm.
This method replaces instances of `await page.waitForSelector('#root');`. page.waitForSelector is discouraged by Playwright.

Reply:
Ah, apologies. I didnt find usages using Ctrl+F, but I think the large diff was hiding that file. Yep, I see 6 instances.